src/surge-xt/gui/overlays/Oscilloscope.cpp

```diff
@@ -1260,19 +1359,23 @@ void Oscilloscope::Background::paintWaveformBackground(juce::Graphics &g)
         std::string minus = "-";
         std::string plus = "+";
         std::stringstream gain;
-        gain << std::fixed << std::setprecision(2) << 1.f / waveform_params_.gain();
+        const unsigned prec = waveform_params_.gain() > 1.f ? 2 : 0;
```

Comment:
I'd explicitly label this as an unsigned int. I think raw unsigned is a deprecated C-ism (here and below).